output/influxdb/output.go

```diff
@@ -209,17 +251,12 @@ func (o *Output) flushMetrics() {
 	o.logger.Debug("Committing...")
 	o.logger.WithField("samples", len(samples)).Debug("Writing...")
 
-	batch, err := o.batchFromSamples(samples)
-	if err != nil {
-		o.logger.WithError(err).Error("Couldn't create batch from samples")
-		return
-	}
+	batch := o.batchFromSamples(samples)
+	o.logger.WithField("points", len(batch)).Debug("Writing...")
 
-	o.logger.WithField("points", len(batch.Points())).Debug("Writing...")
 	startTime := time.Now()
-	if err := o.Client.Write(batch); err != nil {
+	if err := o.pointWriter.WritePoint(context.Background(), batch...); err != nil {
```

Comment:
Have you tried using the Async writer but without first getting all the batches as a whole but just in `batchFromSamples` without creating a whole slice and so on?

Reply:
I tried without using the SampleBuffer, I was pushing directly the received sample to the async writer. In the middle, there was only the tags conversion.